lib/account.py

```diff
@@ -50,13 +79,36 @@ def withdraw(self, amount: int) -> bool:
       return True
     return False
 
-  def calc_gas(self) -> float:
-    # returns gas required to execute contract
-    pass
+  def calc_gas(self, func_name: str) -> float:
+    # calculates the gas cost
+    if func_name in self.funcs:
+      # Redirect the disassemble to a buffer
+      buffer = StringIO()
+      with redirect_stdout(buffer):
+        dis(self.funcs[func_name])
+      
+      # Get the value of disassemble
+      disassemble = buffer.getvalue()
+
+      # calculate the number of instructions
+      num_of_instructions =  len(disassemble.split('\n'))
+
+      # Gas is equal to the number of instructions that need to be executed
+      return num_of_instructions * GAS_PRICE
 
-  def charge_gas(self, gas: float) -> bool:
+    return 0
+
+  def charge_gas(self, gas: float, func_name: str, func_args: tuple) -> bool:
```

Comment:
How are you supposed to charge an external account if it executes a contract? This implementation seems like you're charging a contract for being run.